### PR TITLE
[Auto Parallel] Add doc for DistModel

### PR DESCRIPTION
**Pull request by pkuzyc**
### PR types
Others

### PR changes
APIs

### Description
Pcard-76459
1. Add doc for DistModel.
2. Modify the default value of ``stop_gradient`` arg in ``shard_tensor`` from True to None. Now, if ``stop_gradient`` is not set when using ``shard_tensor``, set the returned Tensor's ``stop_gradient`` identical as the input's stop_gradient when the input has ``stop_gradient`` attribute and True otherwise.

### CODE REVIEW

File: python/paddle/distributed/auto_parallel/api.py [hunk 1]
```diff
@@ -667,11 +729,8 @@ def to_static(
             pipeline parallelism). Default: None.
 
     Returns:
-        DistModel: A DistModel tha contains corresponding computational graph
-            for the input ``layer`` and provides APIs for training, evaluation
-            and prediction.
-        DistributedDataLoader: An optimized data loader that can be used
-            to generate data.
+        DistModel: A ``DistModel`` instance converted the input ``layer``.
+        DistributedDataLoader: An optimized data loader that can be used to generate data.
```
Comment (zhiqiu) [inline python/paddle/distributed/auto_parallel/api.py]: It should be pointed that `DistributedDataLoader` has the same usage with `DataLoader` ?

**Review by jeff41404 — DISMISSED** [review]
LGTM

File: python/paddle/distributed/auto_parallel/api.py (hunk 1 above)
Comment (pkuzyc) [reply python/paddle/distributed/auto_parallel/api.py]: Done

**Review by jzhang533 — APPROVED** [review]
LGTM

**Review by zhiqiu — APPROVED** [review]
LGTM

**Review by jeff41404 — APPROVED** [review]

**Review by sunzhongkai588 — APPROVED** [review]
LGTM for DistModel docs

**Pull request merged**